soc/st/stm32/stm32f1x/CMakeLists.txt

```diff
@@ -1,9 +1,11 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_include_directories(${ZEPHYR_BASE}/drivers)
-zephyr_sources(
-  soc.c
-  )
+zephyr_sources(soc.c)
+
+if(CONFIG_PM OR CONFIG_POWEROFF)
+    zephyr_library_sources(power.c)
```

Comment:
It would be better to split in 2 files.
`z_sys_poweroff` is supposed to be available only if `CONFIG_POWEROFF=y` and same thing for PM hooks with `CONFIG_PM=y` having all them in the same file raise the useless issue of wondering what happens in other cases.
Having 2 files as done on other series is much clearer.